modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2556,3 +2566,63 @@ def _is_trivial_index(cls, index):
             and index.min() == 0
             and index.max() == len(index) - 1
         )
+
+
+class LazyProxyCategoricalDtype(pandas.CategoricalDtype):
```

Comment:
I propose to move this class in to `modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/<metadata/utils>.py`.

Reply:
I think @YarShev meant that the utility file should be called either `metadata.py` or `utils.py`, not that we need to put it into a `metadata/utils.py`, I don't think we want to have a separate directory for this

Reply:
Correct, `metadata.py` sounds better to me.

Reply:
BUT, `utils.py` may have broader usage, we wouldn't need to create a new utility file when we again would need to add some utility code

Reply:
I am also fine with `utils.py`.

Reply:
utils.py seems more reasonable to me either.